---
title: "Introduction"
date: ""
#author:
#- name: "Christina Chatzipantsiou"
#  email: "chatzipantsiou@gmail.com"
output: 
  html_document:
    toc: true
    toc_float: true  
    code_folding: hide
    code_download: true
    theme: cerulean  
    highlight: kate          # specifies the syntax highlighting style
---
<br><br>

![](https://github.com/cgpu/fastQsee_helper_repo/blob/master/images/docker_x_deploit_gif_blogpost_header.gif?raw=true)

<br><br>

# TL;DR

<br>

## Why call it 'fastQsee' though ðŸ¤”?

Well, because you will get to see ðŸ‘€ a FastQC report of your FASTQ file very fast! Installation-less and quick, made possible by a little container magic from Docker and powered by Deploit.

<br>


## What will I learn?

This tutorial is a primer, on how to go modular and use Docker `r emo::ji("whale")` containers for your bioinformatics analysis tasks. More specifically, we will learn how to use the <a href="https://lifebit.page.link/deploit"  target="_blank"><b>Deploit platform</b></a> to assemble and deploy on cloud a reproducible and sharable bioinformatics workflow. <br>

We will assemble the following resources into a workflow on the __Deploit__ platform: <br>

* `r emo::ji("whale")` a __Docker container__, from DockerHub 
* `r emo::ji("rocket")` a __curated pipeline__, available on the __Deploit__ platform
* `r emo::ji("document")` a 1000genomes project __```fastq.gz```__ file, fetched from an EMBL-EBI FTP site

<br>

## Last things first: What results will I get from the FastQC tool on Deploit?

After completing this mini workflow on the Deploit platform, you will have:<br><br> 
__1) a plot-full FastQC html report__, <br>
with key metrics to assess the quality of your FASTQ file <br>

__2) sharable links to your ```Job Page```__ s ,<br> 
with interactive plots and information about the resources and the results.

You can access the  ```Jobs Pages``` from the sharable urls we created for this example: <br>

* for retrieving the 1000genomes fastq.gz file : `r emo::ji('link')` <a href="https://lifebit.page.link/wgetGunzipper__sharable_page"  target="_blank"><code>wgetGunzipper</code></b></a> <br>
* for generating the FastQC html report: `r emo::ji('link')` <a href="https://lifebit.page.link/Job_Page_sharable_fastQsee"  target="_blank"><code>fastQsee</code></b></a><br>

and take a look at what a ```Job Page```  looks like below:

<br><br>
<a href="https://lifebit.page.link/lifebit-deploit" target="_blank"><img src="https://github.com/cgpu/fastQsee_helper_repo/blob/master/images/overview_of_job_page.gif?raw=true" ></a>
<br><br>

Also, you will have the sharable link for the Deploit page job, where you can find the html file of this tutorial (and also the rmarkdown file that created it!). Check below:

<br><br>
<a href="https://lifebit.page.link/fastQsee_Job_Page" target="_blank">
<img src="https://github.com/cgpu/fastQsee_helper_repo/blob/master/images/download_fastqsee_from_deploit.gif?raw=true">
</a>
<br><br>

You can then download them, and you will have a tutorial to guide you on assembling and running reproducible bioinformatics workflows on the Deploit platfom. Now, let's move on with the step by step guide on how to run the FatsQC on Deploit. 

<br>
